lib/koans/10_processes.ex

```diff
@@ -26,15 +45,71 @@ defmodule Processes do
   end
 
   koan "A common pattern is to include the sender in the message, so that it can reply" do
-    pid = spawn(fn -> receive do
-                         {:hello, sender} -> send sender, :how_are_you?
-                      end
-                 end)
+    greeter = fn ->
+      receive do
+        {:hello, sender} -> send sender, :how_are_you?
+      end
+    end
+
+    pid = spawn(greeter)
 
     send pid, {:hello, self()}
     assert_receive ___
   end
 
+  def yelling_echo_loop do
+    receive do
+      {caller, value} ->
+        send caller, String.upcase(value)
+        yelling_echo_loop
+    end
+  end
+
+  koan "Use tail recursion (calling a function as the very last statement) to receive multiple messages" do
+    pid = spawn &yelling_echo_loop/0
+
+    send pid, {self, "o"}
+    send pid, {self, "hai"}
+
+    receive do
```

Comment:
Could we interleave the calls to `send` and `receive` here? So instead of `send-send-receive-receive` we could have `send-receive-send-receive`. Feels more intuitive to me, not sure if you agree.